src/screen/subIdentities/RegisterSubIdentitiesScreen.tsx

```diff
@@ -0,0 +1,182 @@
+import React, {useEffect, useRef, useState} from 'react';
+import {Alert, FlatList, StyleSheet, View} from 'react-native';
+import {Modalize} from 'react-native-modalize';
+import {useQueryClient} from 'react-query';
+import Identicon from '@polkadot/reactnative-identicon';
+import {NavigationProp, RouteProp} from '@react-navigation/native';
+import {Button, Card, Divider, IconProps, Layout, ListItem, Text, TopNavigationAction} from '@ui-kitten/components';
+import Icon from 'presentational/Icon';
+import ModalTitle from 'presentational/ModalTitle';
+import SafeView, {noTopEdges} from 'presentational/SafeView';
+import {useAccountIdentityInfo} from 'src/api/hooks/useAccountIdentityInfo';
+import {useApiTx} from 'src/api/hooks/useApiTx';
+import {SubIdentity, useSubIdentities} from 'src/api/hooks/useSubIdentities';
+import {DashboardStackParamList} from 'src/navigation/navigation';
+import {registerSubIdentitiesScreen} from 'src/navigation/routeKeys';
+import globalStyles, {monofontFamily} from 'src/styles';
+import {AddSubIdentity} from './AddSubIdentity';
+
+type ScreenProps = {
+  navigation: NavigationProp<DashboardStackParamList>;
+  route: RouteProp<DashboardStackParamList, typeof registerSubIdentitiesScreen>;
+};
+
+export function RegisterSubIdentitiesScreen({route, navigation}: ScreenProps) {
+  const address = route.params.address;
+  const modalRef = useRef<Modalize>(null);
+  const {data: identityInfo} = useAccountIdentityInfo(address);
+  const {data: subIdentitiesData} = useSubIdentities(address);
+  const [subIdentities, setSubIdentities] = useState(subIdentitiesData || []);
+  const [submitSubsDisabled, setSubmitSubsDisabled] = useState(true);
+  const queryClient = useQueryClient();
+  const startTx = useApiTx();
+
+  useEffect(() => {
+    if (subIdentitiesData?.length) {
+      setSubIdentities(subIdentitiesData);
```

Comment:
I don't understand why we are not just using `subIdentitiesData` instead of having `subIdentities`?

Reply:
When setting sub identities (making the actual transaction) we send all sub accounts (including the accounts already registered), so we need to keep them in a local state.. When adding and removing sub identities we are just updating the local state so later we can send all accounts in the transaction.

Reply:
got it now ✅